electricitymap/contrib/lib/models/events.py

```diff
@@ -40,9 +41,11 @@ def add_value(
         """
         existing_value: float | None = getattr(self, mode)
         if existing_value is not None:
-            value = 0 if value is None else value
+            value = 0 if value is None or math.isnan(value) else value
             self.__setattr__(mode, existing_value + value)
         else:
+            if value is not None and math.isnan(value):
+                value = None
```

Comment:
Bit unsure if we want to do this but was the simplest way to ensure it was functioning as expected. Would be happy to modify it though.